photon-client/src/main/scala/com/linkedin/photon/ml/hyperparameter/HyperparameterConfigurationType.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.hyperparameter
+
+import com.linkedin.photon.ml.HyperparameterTuningMode.HyperparameterTuningMode
+import com.linkedin.photon.ml.util.DoubleRange
+
+/**
+ * Configurations of hyper-parameters, containing (tuningMode, names, ranges, discreteParams, transformMap).
+ * The Map discreteParams has (index, number of discrete values) as (key, value) pairs. The Map transformMap
+ * has (index, transformation function name) as (key, value) pairs.
+ */
+object HyperparameterConfigurationType {
+  type HyperparameterConfig = (HyperparameterTuningMode, Seq[String], Seq[DoubleRange], Map[Int, Int], Map[Int, String])
```

Comment:
You should make this a case class instead - it'll be easier to pass around and reference the data inside, instead of creating a type short had for a 5 field tuple.